### PR TITLE
Components/checkbox

### PR DESCRIPTION
**Pull request by palingheorghe**
Related to #57.

Props:
- checkboxStyle
-   disabled
-   flexDirection
-   iconColor
-   iconName
-   iconSize
-   iconFamily
-   image
-   imageStyle
-   initialValue
-   label
-   labelStyle
-   onChange
-   styles
-   theme

I'm looking forward to your code review guys! I've already tested this in a different app but if you could test it as well that'd be great! ❤️

🤙🏽

### CODE REVIEW

File: src/Checkbox.js [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+    switch (direction) {
+      case "row":
+        return { marginLeft: 10 };
+      case "row-reverse":
+        return { marginRight: 10 };
+      case "column":
+        return { marginTop: 10 };
+      case "column-reverse":
+        return { marginBottom: 10 };
+      default:
```
Comment (hetmann) [inline src/Checkbox.js]: row & default return the same value

```
case "row":
default:
    return { marginLeft: 10 };
```

File: src/Checkbox.js [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+      flexDirection,
+      image,
+      labelStyle,
+      imageStyle,
+      styles
+    } = this.props;
+
+    const textStyle = [
+      styles.textStyles,
+      disabled && { color: "#707070", opacity: 0.7 },
```
Comment (hetmann) [inline src/Checkbox.js]: disabled color should added to the theme

File: src/Checkbox.js [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+      label,
+      disabled,
+      flexDirection,
+      image,
+      labelStyle,
+      imageStyle,
+      styles
+    } = this.props;
+
+    const textStyle = [
```
Comment (hetmann) [inline src/Checkbox.js]: textStyle => labelStyles

File: src/Checkbox.js [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+    if (!label && !image) return null;
+  }
+
+  // adding the check icon
+  renderChecked() {
+    const { iconName, iconFamily, iconColor, iconSize } = this.props;
+
+    if (this.state.checked)
+      return <Icon name={iconName} family={iconFamily} color={iconColor} size={iconSize} />;
+  }
```
Comment (hetmann) [inline src/Checkbox.js]: By default this should return null

File: src/Checkbox.js [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+      borderWidth: theme.SIZES.BORDER_WIDTH,
+      alignItems: "center",
+      justifyContent: "center",
+      borderRadius: theme.SIZES.BORDER_RADIUS
+    },
+    uncheckedBoxView: {
+      backgroundColor: theme.COLORS.TRANSPARENT,
+      borderColor: theme.COLORS.GREY
+    },
+    checkedBoxView: {
```
Comment (hetmann) [inline src/Checkbox.js]: checkedBoxView => checked

File: src/Checkbox.js [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+      backgroundColor: theme.COLORS.THEME,
+      borderColor: theme.COLORS.THEME
+    },
+    disabledCheckBoxView: {
```
Comment (hetmann) [inline src/Checkbox.js]: disabledCheckBoxView => disabled

File: src/Checkbox.js [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+      flexDirection && this.spaceAround(flexDirection)
+    ];
+    const imageStyles = [
+      styles.imgStyles,
+      imageStyle,
+      flexDirection && this.spaceAround(flexDirection)
+    ];
+
+    if (image && !label)
+      return <Image style={imageStyles} source={{ uri: image }} />;
```
Comment (hetmann) [inline src/Checkbox.js]: What does the Image do? Can you provide an example?

File: src/Checkbox.js (hunk 1 above)
Comment (palingheorghe) [reply src/Checkbox.js]: Do you recommend removing the 'row' case?

Comment (hetmann) [reply src/Checkbox.js]: Maybe row is the default direction

File: src/Checkbox.js (hunk 7 above)
Comment (hetmann) [reply src/Checkbox.js]: Or you can use the children prop from react and have a full customisation

**Pull request merged**